src/main/scala/viper/gobra/frontend/info/implementation/property/Assignability.scala

```diff
@@ -54,13 +54,17 @@ trait Assignability extends BaseProperty { this: TypeInfoImpl =>
       // `var y IntType = x` where x is and int var and IntType is a defined type defined as an int
       // case (l, r) if !(l.isInstanceOf[DeclaredT] && r.isInstanceOf[DeclaredT])
       //  && identicalTypes(underlyingType(l), underlyingType(r)) => true
+      case (l, r) if !(l.isInstanceOf[DeclaredT] && r.isInstanceOf[DeclaredT]) // it does hold for structs
+        && underlyingType(l).isInstanceOf[StructT] && underlyingType(r).isInstanceOf[StructT]
+        && identicalTypes(underlyingType(l), underlyingType(r)) => true
```

Comment:
Even though I said before that this rule was not implemented by the compiler, I was wrong - in fact, the rule holds for all types, not only for structs. Please check https://github.com/viperproject/gobra/issues/133 for a detailed description of the issue.

In a future PR, these checks (`underlyingType(l).isInstanceOf[StructT] && underlyingType(r).isInstanceOf[StructT]`) should be removed.